pkg/agent/agent.go

```diff
@@ -47,6 +48,7 @@ type Agent struct {
 	privateKey sshcert.RawPrivateKey
 
 	hooks map[string]*hook.Hook
+	lock  sync.Mutex
```

Comment:
nit: consider saying what the lock is protecting, both in comment and in descriptive name